gobblin-config-management/gobblin-config-client/src/main/java/gobblin/config/client/ConfigClient.java

```diff
@@ -0,0 +1,360 @@
+   * 1. URI missing authority and configuration store root , for example "etl-hdfs:///datasets/a1/a2". It will get
+   *    the configuration based on the default {@link ConfigStore} in etl-hdfs {@link ConfigStoreFactory}
+   * 2. Complete URI:  for example "etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest/"
+   * 
+   * @return  the resolved {@link Config} based on the input URI.
+   * 
+   * @throws ConfigStoreFactoryDoesNotExistsException: if missing scheme name or the scheme name is invalid
+   * @throws ConfigStoreCreationException: Specified {@link ConfigStoreFactory} can not create required {@link ConfigStore}
+   * @throws VersionDoesNotExistException: Required version does not exist anymore ( may get deleted by retention job )
+   */
+  public Config getConfig(URI configKeyUri) throws ConfigStoreFactoryDoesNotExistsException,
+      ConfigStoreCreationException, VersionDoesNotExistException {
+    ConfigStoreAccessor accessor = this.getConfigStoreAccessor(configKeyUri);
+    ConfigKeyPath configKeypath = ConfigClientUtils.buildConfigKeyPath(configKeyUri, accessor.configStore);
+    return accessor.valueInspector.getResolvedConfig(configKeypath);
+  }
+  
+  /**
+   * batch process for {@link #getConfig(URI)} method
+   * @param configKeyUris
+   * @return
+   * @throws ConfigStoreFactoryDoesNotExistsException
+   * @throws ConfigStoreCreationException
+   * @throws VersionDoesNotExistException
+   */
+  public Map<URI, Config> getConfigs(Collection<URI> configKeyUris)throws ConfigStoreFactoryDoesNotExistsException,
+  ConfigStoreCreationException, VersionDoesNotExistException {
+    if(configKeyUris == null || configKeyUris.size()==0 )
+      return Collections.emptyMap();
+    
+    Map<URI, Config> result = new HashMap<>();
+    Multimap<ConfigStoreAccessor, ConfigKeyPath> partitionedAccessor = ArrayListMultimap.create();
+    Map<ConfigStoreAccessor, URI> inputURI = new HashMap<>();
+    
+    // partitioned the ConfigKeyPaths which belongs to the same store to one accessor 
+    for(URI u: configKeyUris){
+      ConfigStoreAccessor accessor = this.getConfigStoreAccessor(u);
+      ConfigKeyPath configKeypath = ConfigClientUtils.buildConfigKeyPath(u, accessor.configStore);
+      partitionedAccessor.put(accessor, configKeypath);
+      
+      /*
+       * Store the first ConfigKeyURI to the map, and the return format is based on it
+       */
+      if(!inputURI.containsKey(accessor)){
+        inputURI.put(accessor, u);
+      }
+    }
+    
+    for(Map.Entry<ConfigStoreAccessor, Collection<ConfigKeyPath>> entry: partitionedAccessor.asMap().entrySet()){
+      Map<ConfigKeyPath, Config> batchResult= entry.getKey().valueInspector.getResolvedConfigs(entry.getValue());
+      // translate the ConfigKeyPath to URI
+      for(Map.Entry<ConfigKeyPath, Config> resultEntry: batchResult.entrySet()){
+        URI formattedUri = ConfigClientUtils.buildURI(resultEntry.getKey(), inputURI.get(entry.getKey()),
```

Comment:
See my comment below about. I don't understand at a high level what buildURI() is supposed to do. This method seems to be overly complicated. I think it is OK to say that getConfigs() works only for configs coming from the same store/accessor. This will simplify the implementation significantly. You can just keep a reverse map from a ConfigKeyPath to a ConfigKeyURI that was used to generate it. Then populating the result is done using this map.